File: CarSpecAPI/Data/Models/DataExtractionModel/AuditChangeDto.cs
using System.Text.Json;

namespace CarSpecAPI.Data.Models.DataExtractionModel
{
    public class AuditChangeDto
    {
        public string ChangeId { get; set; } = string.Empty;

        public string IssueType { get; set; } = string.Empty;

        public string Category { get; set; } = string.Empty;

        public string Severity { get; set; } = string.Empty;

        public string Action { get; set; } = string.Empty;

        public AuditChangeTargetDto Target { get; set; } = new();

        public JsonElement? Data { get; set; }

        public int? PageNumber { get; set; }

        public string? Evidence { get; set; }

        public string? Reason { get; set; }
    }
}

File: CarSpecAngular/carspec/src/app/core/models/interfaces/DataProcessingDtos/apiResponseDtos/audit-change-dto.ts
import { AuditChangeTargetDto } from "./audit-change-target-dto";

export interface AuditChangeDto {
      changeId: string;
  issueType: string;
  category: string;
  severity: string;
  action: string;
  target: AuditChangeTargetDto;
  data: any;
  pageNumber: number | null;
  evidence: string | null;
  reason: string | null;
}
